File: c++/src/semaine/datatypes/xml/SemaineML.cpp
/*
 *  SemaineML.cpp
 *  semaine
 *
 *  Created by Marc Schröder on 06.11.08.
 *  Copyright 2008 DFKI GmbH. All rights reserved.
 *
 */

#include "SemaineML.h"

namespace semaine {
namespace datatypes {
namespace xml {

const std::string SemaineML::version = "0.0.3";
const std::string SemaineML::namespaceURI = "http://www.semaine-project.eu/semaineml";

	// Elements
const std::string SemaineML::E_SETUP = "setup";
const std::string SemaineML::E_CONTEXT = "situational-context";
const std::string SemaineML::E_USERSTATE = "user-state";
const std::string SemaineML::E_AGENTSTATE = "agent-state";
const std::string SemaineML::E_DIALOGSTATE = "dialog-state";
const std::string SemaineML::E_FEATURE = "feature";
const std::string SemaineML::E_EVENT = "event";
const std::string SemaineML::E_BEHAVIOUR = "behaviour";
const std::string SemaineML::E_TEXT = "text";
const std::string SemaineML::E_EMOTIONALLY_CONCORDANT_WITH_USER = "emotionally-concordant-with-user";
const std::string SemaineML::E_DIALOG_ACT = "dialog-act";
const std::string SemaineML::E_DIALOG_HISTORY = "dialog-history";
const std::string SemaineML::E_SPEAKER = "speaker";
const std::string SemaineML::E_LISTENER = "listener";
const std::string SemaineML::E_TOPIC = "topic";
const std::string SemaineML::E_USER = "user";
const std::string SemaineML::E_CHARACTER = "character";
const std::string SemaineML::E_SPEAKING = "speaking";

	// Attributes
const std::string SemaineML::A_WHO = "who";
const std::string SemaineML::A_NAME = "name";
const std::string SemaineML::A_SPEAKER = "speaker";
const std::string SemaineML::A_TOPIC = "topic";
const std::string SemaineML::A_VALUE = "value";
const std::string SemaineML::A_TIME = "time";
const std::string SemaineML::A_INTENSITY = "intensity";
const std::string SemaineML::A_STATUS = "status";
const std::string SemaineML::A_STATUS_CHANGE = "statusChange";

// Attribute values
// For A_STATUS:
const std::string SemaineML::V_PRESENT = "present";
const std::string SemaineML::V_ABSENT = "absent";


} // namespace xml
} // namespace datatypes
} // namespace semaine

File: java/src/eu/semaine/datatypes/xml/SemaineML.java
/**
 * Copyright (C) 2008 DFKI GmbH. All rights reserved.
 * Use is subject to license terms -- see license.txt.
 */
package eu.semaine.datatypes.xml;

/**
 * @author marc
 *
 */
public class SemaineML
{
    public static final String version = "0.0.3";
    public static final String namespaceURI = "http://www.semaine-project.eu/semaineml";

    // Elements
    public static final String E_CONTEXT = "situational-context";
    public static final String E_USERSTATE = "user-state";
    public static final String E_AGENTSTATE = "agent-state";
    public static final String E_DIALOGSTATE = "dialog-state";
    public static final String E_SETUP = "setup";
    public static final String E_FEATURE = "feature";
    public static final String E_EVENT = "event";
    public static final String E_BEHAVIOUR = "behaviour";
    public static final String E_TEXT = "text";
    public static final String E_EMOTIONALLY_CONCORDANT_WITH_USER = "emotionally-concordant-with-user";
    public static final String E_DIALOG_ACT = "dialog-act";
    public static final String E_DIALOG_HISTORY = "dialog-history";
    public static final String E_SPEAKER = "speaker";
    public static final String E_LISTENER = "listener";
    public static final String E_TOPIC = "topic";
    public static final String E_USER = "user";
    public static final String E_CHARACTER = "character";
    public static final String E_SPEAKING = "speaking";
    public static final String E_PITCH = "pitch";
    public static final String E_FACEPRESENT = "face-present";
    public static final String E_VOCALIZATION = "vocalization";
    
    // Attributes
    public static final String A_WHO = "who";
    public static final String A_NAME = "name";
    public static final String A_SPEAKER = "speaker";
    public static final String A_TOPIC = "topic";
    public static final String A_VALUE = "value";
    public static final String A_TIME = "time";
    public static final String A_INTENSITY = "intensity";
    public static final String A_STATUS = "status";
    public static final String A_DIRECTION = "direction";
    public static final String A_ID = "id";
    public static final String A_DATA = "data";
    public static final String A_TYPE = "type";
    
    // Attribute values
    // For A_STATUS:
    public static final String V_PRESENT = "present";
    public static final String V_ABSENT = "absent";
    // For A_TYPE
    public static final String V_READY = "ready";
    public static final String V_DELETED = "deleted";
    public static final String V_START = "start";
    public static final String V_STOPPED = "stopped";
    public static final String V_END = "end";
    // For <vocalization name="...">:
    public static final String V_LAUGHTER = "(laughter)";
    public static final String V_SIGH = "(sigh)";
    
}
